Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -127,6 +127,42 @@ void OpenALStream::Clear(bool mute)
   }
 }
 
+static ALenum CheckALError(const char* desc)
+{
+  ALenum err = alGetError();
+
+  if (err != AL_NO_ERROR)
+  {
+    std::string type;
+
+    switch (err)
```

Comment:
It seems that this can be done with `std::map` as seen [here](http://thispointer.com/stdmap-tutorial-part-1-usage-detail-with-examples/).
I can try to implement it if everyone believe it is better.